### DISCUSSION TITLE
23.1.1 Smart commit mode per connection type not in changelog?

### DISCUSSION THREAD

**Discussion by wheinen** [dbeaver/dbeaver#20386] [General]
I just created a new connection after updating to 23.1.1 and I was surprised Smart Commit mode wasn't enabled by default, as I had set it in Preferences > Connections > Transactions.
Now it points to connection types, where you can enable smart commit mode by default per connection type.

While this is a good change, the behaviour of DBeaver was suddenly changed without my knowledge. Luckily I caught it in time, but this is a change that should be in the [changelog](https://dbeaver.io/2023/06/26/dbeaver-23-1-1/) at the very least! I might suggest adding a warning about this change. Or perhaps enabling smart commit by default on all connection types if you had the checkbox 'smart commit' enabled on the old location?

**Comment by wheinen** [comment]
Update: Enabling smart commit per connection type DOES NOT automatically enable smart commit! I can't find a way to enable smart commit by default on new connections now. This is a real issue for our workflow!

**Reply by LonwoLonwo** [reply]
Hello @wheinen 

Could you please check the [latest DBeaver](https://dbeaver.io/download/) version? I hope we fixed it.

**Comment by wheinen** [comment]
Hello @LonwoLonwo 
I installed the latest version, and smart commit is working by default now. Thank you very much! 😄

**Comment by wheinen** [comment]
Hello @LonwoLonwo 

I don't know how I tested this yesterday, but I got a message from my team this function is not working correctly, and when I test it now it isn't working for me as well.

Settings: 
Default connection type for new connections: Development
Connection type setting:
![image](https://github.com/dbeaver/dbeaver/assets/137804655/b8772749-4dcb-4dd8-b923-f3423f380949)

How we add new connectiions:
dbeaver-cli.exe -con "driver=mysql|autoCommit=true|host={HOSTNAME}|port={PORT}|user={USER}|database={DATABASE}|password={PASSWORD}|name={NAME}"

This results in a new connection without smart commit. Isn't there a setting for the CLI that tells the connection to use smart commit?

**Comment by wheinen** [comment]
I created a new issue for this problem: https://github.com/dbeaver/dbeaver/issues/21209
